### DISCUSSION TITLE
Logs of HC ?

### DISCUSSION THREAD

**Discussion by R-Nabil** [healthchecks/healthchecks#902] [Q&A]
Hi,

I'm trying to integrate into crowdsec (ids/ips engine) a scenario to detect any Brute Force attempt to login into HC.
But I can't figure out where are the HC logs output in a selfhosted environment.

Can anyone help ?
Thanks

**Comment by cuu508** [comment]
Are you looking for access logs, where you would see a single IP hitting `/accounts/login/` a lot of times?

Where you get the logs would depend on your self-hosted environment. For example, if you are using nginx as a reverse proxy, you would configure access logs in nginx.

**Comment by R-Nabil** [comment]
Not really. I’m looking for authentication failure logs. Restricting an IP
based on a number of hit could create false positives I believe.

**Reply by cuu508** [reply]
Healthchecks does not emit authentication failure events to logs.

**Comment by R-Nabil** [comment]
So you reckon there is no way to prevent a bot from brute forcing or enumerating users ?
________________________________

**Reply by cuu508** [reply]
Regarding brute-forcing, Healthchecks has built-in rate-limiting, you can see the limits here: https://github.com/healthchecks/healthchecks/blob/527f361a9a302c53d2ad7e3337e573bccae22743/hc/api/models.py#L1127 

Regarding enumeration, the signup and login forms return the same response for registered and unregistered email addresses: https://github.com/healthchecks/healthchecks/security/advisories/GHSA-frxj-v529-jv6w

**Comment by R-Nabil** [comment]
Does that rate-limiting emit a log ? I'd like to blacklist IP who would hit the rate-limit

**Reply by cuu508** [reply]
No, rate-limiting does not emit logs.
